src/character_training.py

```diff
@@ -137,21 +176,57 @@ def extract_feature_vector(filename):
 
 
 def train_character_svm(features, labels):
+    # crossvalidation
+    cv = ShuffleSplit(n_splits=5, test_size=0.2, random_state=7)
+
+    # paramgrid
+    param_grid=[{'C': [2**x for x in config.C_RANGE]}]
+
+    # model
     model = LinearSVC(penalty='l2',
                       loss='squared_hinge',
                       dual=True,
                       tol=0.0001,
-                      C=1.0,
                       multi_class='ovr',
                       fit_intercept=True,
                       intercept_scaling=1,
                       class_weight=None,
                       verbose=0,
                       random_state=None,
                       max_iter=1000)
+
+    # gridsearch
+    start = time.time()
+    classifier = GridSearchCV(estimator=model,
+                              cv=cv,
+                              param_grid=param_grid,
+                              refit=True,
+                              verbose=3,
+                              n_jobs=8)
+    classifier.fit(features, labels)
+    end = time.time()
+
+    logging.info("GridSearch done, time: {t}s".format(t = end - start))
+
+    best_C = classifier.best_params_['C']
+    best_model = CalibratedClassifierCV(LinearSVC(LinearSVC(
+        penalty='l2',
+        loss='squared_hinge',
+        dual=True,
+        tol=0.0001,
+        C=best_C,
+        multi_class='ovr',
+        fit_intercept=True,
+        intercept_scaling=1,
+        class_weight=None,
```

Comment:
can we assure that the classes are balanced? do all characters have a sufficient number of samples?